docs/manual/mustcall-checker.tex

```diff
@@ -0,0 +1,184 @@
+\htmlhr
+\chapterAndLabel{Must Call Checker}{mustcall-checker}
+
+The Must Call Checker conservatively over-approximates
+the set of methods that an object should call before it is de-allocated.
+It is intended to be run as a subchecker of another checker: on its own, it does not
+enforce any rules other than subtyping.
+
+For example, consider an object whose type is \<java.io.OutputStream>.
+This \<OutputStream> might have an obligation to call the \<close()> method, to release an
+underlying file resource. The type of this \<OutputStream> is \<@MustCall(\{"close"\})>.
+ Or, the \<OutputStream> might not have such an obligation, if the
+underlying resource is a byte array. The type of this \<OutputStream> is \<@MustCall(\{\})>.
+The Must Call Checker over-approximates the methods that an arbitrary
+\<OutputStream> should call by assigning it the type \<@MustCall(\{"close"\}) OutputStream>, which can be read
+as "an OutputStream that might need to call close() (but no other methods) before it is de-allocated".
+
+\sectionAndLabel{Must Call annotations}{must-call-annotations}
+
+The Must Call Checker supports these type qualifiers, whose subtyping relationships
+are shown in Figure~\ref{fig-must-call-hierarchy}:
+
+\begin{description}
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{MustCall}{String[] value}]
+  gives a superset of the methods that the annotated program element
+  must call before it is de-allocated.
+  The default type qualifier for an unannotated type is \<@MustCall(\{\})>.
+
+\item[\refqualclass{checker/mustcall/qual}{MustCallUnknown}]
+  represents a value with an unknowable or infinite set of must-call methods.
+  It is used internally by the type system but should never be written by a
+  programmer.
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{InheritableMustCall}{String[] value}]
+  is an alias of \<@MustCall> that can only be written on a class declaration.
+  It applies both to the class declaration on which it is written and also to all subclasses.
+  This frees the user of the need to write the \<@MustCall> annotation on every subclass.
+
+\item[\refqualclass{checker/mustcall/qual}{PolyMustCall}]
+  is the polymorphic annotation for the Must Call type system.
+  For a description of polymorphic qualifiers, see
+  Section~\ref{method-qualifier-polymorphism}.
+
+\end{description}
+
+\begin{figure}
+\includeimage{mustcall}{5cm}
+\caption{Subtyping relationships between examples of type qualifiers from the Must Call Checker's type
+hierarchy (note that this is only part of the full hierarchy, which contains one \<@MustCall> annotation
+for every combination of methods).
+Qualifiers in gray are used internally by the type system but should
+never be written by a programmer.}
+\label{fig-must-call-hierarchy}
+\end{figure}
+
+\<@MustCall(\{"foo"\}) Object> can only
```

Comment:
Somewhat, but I'd rather keep both since they explain the concept in different ways, which might help a reader who doesn't understand or is confused after only reading one of the two.